test/saithrift/tests/sail2.py

```diff
@@ -456,3 +456,76 @@ def runTest(self):
+        switch_init(self.client)
+        vlan_id = 10
+        port1 = port_list[0]
+        port2 = port_list[1]
+        port3 = port_list[2]
+        mac1 = '00:11:11:11:11:11'
+        mac2 = '00:22:22:22:22:22'
+        fdb_aging_time = 30
+
+        vlan_oid = sai_thrift_create_vlan(self.client, vlan_id)
+        vlan_member1 = sai_thrift_create_vlan_member(self.client, vlan_oid, port1, SAI_VLAN_TAGGING_MODE_UNTAGGED)
+        vlan_member2 = sai_thrift_create_vlan_member(self.client, vlan_oid, port2, SAI_VLAN_TAGGING_MODE_UNTAGGED)
+        vlan_member3 = sai_thrift_create_vlan_member(self.client, vlan_oid, port3, SAI_VLAN_TAGGING_MODE_UNTAGGED)
+
+        attr_value = sai_thrift_attribute_value_t(u16=vlan_id)
+        attr = sai_thrift_attribute_t(id=SAI_PORT_ATTR_PORT_VLAN_ID, value=attr_value)
+        self.client.sai_thrift_set_port_attribute(port1, attr)
+        self.client.sai_thrift_set_port_attribute(port2, attr)
+        self.client.sai_thrift_set_port_attribute(port3, attr)
+
+        attr_value = sai_thrift_attribute_value_t(u32=fdb_aging_time)
+        attr = sai_thrift_attribute_t(id=SAI_SWITCH_ATTR_FDB_AGING_TIME, value=attr_value)
+        self.client.sai_thrift_set_switch_attribute(attr)
+
+        pkt = simple_tcp_packet(eth_dst=mac2,
+                                eth_src=mac1,
+                                ip_dst='10.0.0.1',
+                                ip_id=101,
+                                ip_ttl=64)
+
+        pkt1 = simple_tcp_packet(eth_dst=mac1,
+                                 eth_src=mac2,
+                                 ip_dst='10.0.0.1',
+                                 ip_id=101,
+                                 ip_ttl=64)
+
+        try:
+            print "Send packet from port1 to port2 and verify on each of ports"
+            print '#### Sending 00:11:11:11:11:11| 00:22:22:22:22:22 | 10.10.10.1 | 192.168.0.1 | @ ptf_intf 1 ####'
+            send_packet(self, 0, str(pkt))
+            verify_each_packet_on_each_port(self, [pkt, pkt], [1, 2])
+            print "Send packet from port2 to port1 and verify only on port1"
+            print '#### Sending 00:22:22:22:22:22| 00:11:11:11:11:11 | 10.10.10.1 | 192.168.0.1 | @ ptf_intf 2 ####'
+            send_packet(self, 1, str(pkt1))
+            verify_packets(self, pkt1, [0])
+            print "Wait when the aging time for FDB entries in the FDB table expires, and the entries are removed ..."
+            time.sleep(fdb_aging_time + 5)
+            print "Send packet from port2 to port1 and verify on each of ports"
+            print '#### Sending 00:22:22:22:22:22| 00:11:11:11:11:11 | 10.10.10.1 | 192.168.0.1 | @ ptf_intf 2 ####'
+            send_packet(self, 1, str(pkt1))
+            verify_each_packet_on_each_port(self, [pkt1, pkt1], [0, 2])
+            print "Wait when the aging time for FDB entries in the FDB table expires, and the entries are removed ..."
+            time.sleep(fdb_aging_time + 5)
```

Comment:
my comment was about the 2nd sleep and not on the 1st sleep
the 1st sleep is needed so the entries expire, and to be able to functionally test that, as you do, that the entries expired and now flood is happening
the 2nd sleep you use just as a clean up mechanism before the test end, there is no more traffic going on in the test after the 2nd sleep, just clean up, so my suggestion was that if the intention is just to clean up the dynamic entries we might as well just flush them instead of waiting for aging, as we already proved the aging is working